lib/spack/spack/database.py

```diff
@@ -355,14 +372,34 @@ def __init__(self, root, db_dir=None, upstream_dbs=None,
         # message)
         self._fail_when_missing_deps = False
 
+        self._write_transaction_impl = lk.WriteTransaction
+        self._read_transaction_impl = lk.ReadTransaction
+        self._record_fields = default_install_record_fields
+
+    @contextlib.contextmanager
+    def enable_buildcache_index_mode(self):
```

Comment:
I think this should be `def change_mode(self, write_lock_impl, read_lock_impl, include_fields)` to make it a general context manager. And then the build cache can control which options it passes to the context manager, which seems like a more appropriate place to store the information about what fields the build cache needs.

Reply:
Yeah makes sense, will do.  Thanks for the review.